bpf-gpl/tc.c

```diff
@@ -611,6 +611,12 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 			}
 		}
 	}
+	if (state.ip_proto == IPPROTO_ICMP)
+	{
+		state.icmp_type = icmp_header->type;
```

Comment:
Any reason to leave this initialisation so late?  We initialise state->sport/dport all the way in the switch at line 465.  I'd be inclined to do the initialisation in that switch for symmetry.

Reply:
We do conntrack lookup based on state->sport, dport. Since we are not clear with conntrack for icmp, i have added it before we jump into policy

Reply:
Ah, worth a comment then so someone doesn't make the mistake I made.  I'd probably say something like
```
// icmp_type and icmp_code share storage with the ports; now we've used
// the ports set to 0 to do the conntrack lookup, we can set the ICMP fields 
// for policy.
```